src/main/java/com/ays/assignment/service/impl/AssignmentServiceImpl.java

```diff
@@ -95,6 +96,28 @@ public AysPage<Assignment> getAssignments(AssignmentListRequest listRequest) {
         );
     }
 
+    /**
+     * <p>Retrieves an assignment of a user if there's one with the statuses:</p>
+     * <li>RESERVED</li>
+     * <li>ASSIGNED</li>
+     * <li>IN_PROGRESS</li>
+     * <p>If there's no assignment with the above statuses, the method throws {@link AysAssignmentNotExistByIdException}.</p>
+     *
+     * @return Assignment
+     */
+    @Override
+    public Assignment getAssignmentSummary() {
+
+        String userId = identity.getUserId();
+
+        AssignmentEntity assignmentEntity = assignmentRepository.findByUserIdAndStatusNot(userId, AssignmentStatus.DONE)
+                .filter(entity -> !entity.isAvailable())
```

Comment:
`entity` demek yerine `assignmentEntityFromDatabase` veya `assignment` gibi bir isimlendirme yaparsak daha okunaklı olabilir.

Reply:
Ek olarak filter kısmında statüleri filtrelememişiz, tam olarak nasıl bir kontrol yapıyoruz? Yarın farklı bir statü eklenirse buradaki kontrol bozulmuş olacak.

Reply:
Haklısın, burası için farklı bir repository metodu uygulacağım.

Reply:
Hallettim.

Reply:
@AhmetAksunger anladığım kadarıyla @agitrubard aşağıdaki gibi bir kullanımdan bahsediyor. @agitrubard doğru anladım değil mi?
```java
AssignmentEntity assignmentEntity = assignmentRepository.findByUserIdAndStatusNot(userId, AssignmentStatus.DONE)
                .filter(AssignmentEntity:isReserved)
                .filter(AssignmentEntity:isInProgress)
                .filter(AssignmentEntity:isAssigned)
                .orElseThrow(
                        () -> new AysAssignmentNotExistByUserIdAndStatusException(userId, AssignmentStatus.RESERVED, AssignmentStatus.ASSIGNED, AssignmentStatus.IN_PROGRESS)
                );
```

Reply:
Verdiğin örnek doğru değil. Assignment statüsü ASSIGNED olan bir kullanıcı düşünelim. İlk filtre isReserved olduğu için ve filtreyi sağlamadığı için Optional.empty dönülecek ve diğer filtreler Optional.empty'yi filtrelemeye çalışacak işin sonunda da orElseThrow metodu çalışacak hiçbir filtre tutmadığı için.
Verdiğin örneğin düzgün hali şu:
```java
AssignmentEntity assignmentEntity = assignmentRepository.findByUserIdAndStatusNot(userId, AssignmentStatus.DONE)
                .filter(e -> e.isReserved || e.isInProgress || e.isAssigned)
                .orElseThrow(
                        () -> new AysAssignmentNotExistByUserIdAndStatusException(userId, AssignmentStatus.RESERVED, AssignmentStatus.ASSIGNED, AssignmentStatus.IN_PROGRESS)
                );
```
Fakat ben bu işlemi filter içinde halletmek yerine `findByUserIdAndStatusIn` repository metoduyla hallettim. Hem SQL sorgusu atılacağı için daha hızlı çalışacak, hem de `acceptedStatuses` diye bir set oluşturulduğu için okunurluğun daha yüksek olduğunu düşünüyorum.
```java
    @Override
    public Assignment getAssignmentSummary() {

        final EnumSet<AssignmentStatus> acceptedStatuses = EnumSet.of(
                AssignmentStatus.ASSIGNED, AssignmentStatus.RESERVED, AssignmentStatus.IN_PROGRESS
        );

        String userId = identity.getUserId();

        AssignmentEntity assignmentEntity = assignmentRepository.findByUserIdAndStatusIn(userId, acceptedStatuses)
                .orElseThrow(() -> new AysAssignmentNotExistByUserIdAndStatusException(userId, acceptedStatuses));
        return assignmentEntityToAssignmentMapper.map(assignmentEntity);
    }
```

Reply:
@ibrahimdenizz @AhmetAksunger elinize sağlık. Doğrudan veritabanından çekmemiz daha iyi olabilir yani son örnekte verdiğin gibi `...StatusIn` olarak kullanalım ama kod yazımı olarak aşağıdaki gibi yapalım;

```java
@Override
public Assignment getAssignmentSummary() {

    final String userId = identity.getUserId();

    final EnumSet<AssignmentStatus> acceptedStatuses = EnumSet.of(
            AssignmentStatus.ASSIGNED, AssignmentStatus.RESERVED, AssignmentStatus.IN_PROGRESS
   );

    final AssignmentEntity assignmentEntity = assignmentRepository.findByUserIdAndStatusIn(userId, acceptedStatuses)
         .orElseThrow(() -> new AysAssignmentNotExistByUserIdAndStatusException(userId, acceptedStatuses));
    return assignmentEntityToAssignmentMapper.map(assignmentEntity);
}
```

Reply:
Hallettim.